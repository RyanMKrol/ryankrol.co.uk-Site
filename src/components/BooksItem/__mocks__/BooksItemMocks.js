import BooksItem from '../BooksItem';

const DEFAULT_MOCK = (
  <BooksItem
    authors={['Matt Haig']}
    images={{
      thumbnail:
        'http://books.google.com/books/content?id=4mZ1XoSpR9UC&printsec=frontcover&img=1&zoom=1&source=gbs_api',
    }}
    isbn={'9780857868770'}
    series={'The Humans'}
    numberInSeries={1}
    title="The Humans"
  />
);

export default DEFAULT_MOCK;
