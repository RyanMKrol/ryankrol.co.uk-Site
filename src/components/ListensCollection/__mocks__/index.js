import ListensCollection from '../ListensCollection';

const FULL_OUTPUT_MOCK = <ListensCollection outputSize="full" />;
const REDUCED_OUTPUT_MOCK = <ListensCollection outputSize="reduced" />;

const EXAMPLE_LISTENS_MOCK_ONE = {
  albumLink: 'test-albumLink-one',
  albumName: 'test-albumName-one',
  thumbnail: 'test-thumbnail-one',
  artist: 'test-artist-one',
  playcount: 1,
};

const EXAMPLE_LISTENS_MOCK_TWO = {
  albumLink: 'test-albumLink-two',
  albumName: 'test-albumName-two',
  thumbnail: 'test-thumbnail-two',
  artist: 'test-artist-two',
  playcount: 2,
};

export {
  FULL_OUTPUT_MOCK,
  REDUCED_OUTPUT_MOCK,
  EXAMPLE_LISTENS_MOCK_ONE,
  EXAMPLE_LISTENS_MOCK_TWO,
};
