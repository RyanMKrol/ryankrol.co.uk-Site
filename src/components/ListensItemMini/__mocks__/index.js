import ListensItemMini from '../ListensItemMini';

const DEFAULT_MOCK = (
  <ListensItemMini
    albumLink="https://www.last.fm/music/Kendrick+Lamar/DAMN."
    albumName="DAMN"
    thumbnail="https://lastfm.freetls.fastly.net/i/u/300x300/243a7444e2d59ff9b38e123f7267e3be.jpg"
  />
);

export default DEFAULT_MOCK;
