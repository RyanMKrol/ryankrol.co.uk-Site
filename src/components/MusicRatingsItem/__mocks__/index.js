import MusicRatingsItem from '../MusicRatingsItem';

const DEFAULT_MOCK = (
  <MusicRatingsItem
    highlights={'Selah, Follow God'}
    artist={'Kanye West'}
    mood={'Vibin'}
    date={'16-05-2021'}
    rating={71}
    title={'JESUS IS KING'}
  />
);

export default DEFAULT_MOCK;
