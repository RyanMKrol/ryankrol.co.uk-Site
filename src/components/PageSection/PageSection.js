import React from 'react';

import styles from './PageSection.module.css';

/**
 * Generates the section title
 *
 * @param {string} title The title of the section
 */
function generateSectionTitle(title) {
  return title ? <h1 className={styles.title}>{title}</h1> : null;
}

/**
 * Component for rendering page section
 *
 * @param {object} props The props
 * @returns {React.Component} JSX PageSection component
 */
function PageSection(props) {
  return (
    <div className="section">
      {generateSectionTitle(props.title)}
      <div className="page-container">{props.children}</div>
    </div>
  );
}

export default PageSection;
