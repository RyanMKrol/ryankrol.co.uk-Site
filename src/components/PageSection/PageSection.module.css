.title {
  opacity: 0.15;
  font-size: 150px;
  user-select: none;
}
