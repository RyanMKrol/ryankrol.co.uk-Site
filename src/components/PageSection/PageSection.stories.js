import React from 'react';

import { MOCK_WITH_TITLE, MOCK_WITHOUT_TITLE } from './__mocks__';

export default {
  title: 'Components/PageSection',
};

/**
 * Section without a title
 *
 * @returns {React.Component} JSX for the PageSection
 */
export const SectionWithoutTitle = () => MOCK_WITHOUT_TITLE;

/**
 * Section with a title
 *
 * @returns {React.Component} JSX for the PageSection
 */
export const SectionWithTitle = () => MOCK_WITH_TITLE;
