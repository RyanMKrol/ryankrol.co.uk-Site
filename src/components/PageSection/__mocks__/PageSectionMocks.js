import PageSection from '../PageSection';

const MOCK_WITH_TITLE = (
  <PageSection title="Example">
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit,
      <span className="highlighted">sed do eiusmod tempor incididunt ut</span> labore et dolore
      magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
      aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit
      esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident,
      sunt in culpa qui officia deserunt mollit anim id est laborum
    </p>
  </PageSection>
);

const MOCK_WITHOUT_TITLE = (
  <PageSection>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit,
      <span className="highlighted">sed do eiusmod tempor incididunt ut</span> labore et dolore
      magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
      aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit
      esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident,
      sunt in culpa qui officia deserunt mollit anim id est laborum
    </p>
  </PageSection>
);

export { MOCK_WITH_TITLE, MOCK_WITHOUT_TITLE };
