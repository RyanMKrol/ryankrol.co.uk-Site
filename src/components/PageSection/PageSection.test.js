import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import {
  render, screen, fireEvent, waitFor,
} from '@testing-library/react';

import { BrowserRouter as Router } from 'react-router-dom';
import PageSection from './PageSection';

import { MOCK_WITH_TITLE, MOCK_WITHOUT_TITLE } from './__mocks__/PageSectionMocks';

it('renders page section correctly without a title', () => {
  const pageSection = shallow(MOCK_WITHOUT_TITLE);
  expect(toJson(pageSection)).toMatchSnapshot();
});

it('renders page section correctly with a title', () => {
  const pageSection = shallow(MOCK_WITH_TITLE);
  expect(toJson(pageSection)).toMatchSnapshot();
});

it('renders page section without a title without throwing', () => {
  expect(() => mount(MOCK_WITHOUT_TITLE)).not.toThrow();
});

it('renders page section with a title without throwing', () => {
  expect(() => mount(MOCK_WITH_TITLE)).not.toThrow();
});
