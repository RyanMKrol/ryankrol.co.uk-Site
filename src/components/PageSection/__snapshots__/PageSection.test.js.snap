// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders page section correctly with a title 1`] = `
<div
  className="section"
>
  <h1
    className="title"
  >
    Example
  </h1>
  <div
    className="page-container"
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit,
      <span
        className="highlighted"
      >
        sed do eiusmod tempor incididunt ut
      </span>
       labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum
    </p>
  </div>
</div>
`;

exports[`renders page section correctly without a title 1`] = `
<div
  className="section"
>
  <div
    className="page-container"
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit,
      <span
        className="highlighted"
      >
        sed do eiusmod tempor incididunt ut
      </span>
       labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum
    </p>
  </div>
</div>
`;
